### PR TITLE
Add nyaa.si and nyaa.pantsu.cat crawlers (Redo of #76)

### PR DESCRIPTION
**Pull request by skwerlman**
This PR adds a crawler for Nyaa.si and nyaa.pantsu.cat.

By default, torrents from sukebei.nyaa.si and sukebei.pantsu.cat are not scraped, but these sections can be uncommented to enable it.

This also fixes a couple problems I ran into while writing the parser:
- Fix failing test caused by missing entry in `valid_attrs`
- Fix a few warnings when running tests
- Don't crash when displaying torrents that were scraped before `outbound_url` was introduced

This PR is based on the `feature/general-cleanup` branch.

Known issues:
- none

### CODE REVIEW

**Comment by pahakalle** [comment]
For those that don't know what sukebei.nyaa.si is. Its the NSFW section of nyaa.si

File: mix.exs [hunk 1]
```diff
@@ -39,13 +39,14 @@ defmodule Magnetissimo.Mixfile do
...
-     {:httpoison, "~> 0.10.0"},
+     {:floki, "~> 0.17.2"},
+     {:httpoison, "~> 0.12"},
      {:html_entities, "~> 0.3"},
      {:distillery, "~> 1.4"},
      {:scrivener_ecto, "~> 1.0"},
      {:scrivener_html, "~> 1.1"},
-     {:sizeable, "~> 0.1.5"}]
+     {:sizeable, "~> 0.1.5"},
+     {:credo, "~> 0.8", only: [:dev, :test], runtime: false}]
```
Comment (skwerlman) [inline mix.exs]: the addition of this and the change of `httpoison`'s version were accidental; i'll revert when i can work on this again

**Comment by pahakalle** [comment]
They might make a little edit to to rss layout in nyaa (https://github.com/nyaadevs/nyaa/commit/6b71cf101ec2133fba035c69c03080de9a812417)

**Comment by skwerlman** [comment]
rather than a little edit in the rss, I can use that new field to get all of the needed data in one pass, rather than just gathering links to the torrent page. this'd make the scraper ~75x faster than it currently is, so thanks for the heads up lol

**Comment by skwerlman** [comment]
actually, thanks to you getting me to look at the rss template, i realised there's another parameter `m` which allows me to get the magnet links directly in the feed. this will let me get all the needed info in one pass over the feed instead of gathering`/view/` links, which'll speed up the scraper ~75x.

**Comment by skwerlman** [comment]
in addition to working on the above changes, i'm also adding support for nyaa.pantsu.cat, another nyaa.se clone.

they have an rss feed as well, but to get the magnet links in the feed, i'm having to use their eztv-compat feed until they add proper support in the regular feed

**Comment by skwerlman** [comment]
This PR is ready to be reviewed.

**Comment by pahakalle** [comment]
Why add nyaa.pantsu.cat?

**Comment by skwerlman** [comment]
the database is not shared with nyaa.si's, so there are different torrents on each

**Comment by skwerlman** [comment]
@sergiotapia is there any other work this PR needs?

**Comment by tchoutri** [comment]
@skwerlman Hi! Could you see if your PR still works with the new code in `master`? I made some changes, especially by switching to Elixir 1.5. If it works without fuss I'd be glad to merge it. :)

**Comment by skwerlman** [comment]
sure, i'll check it in a couple minutes. I think i'll need to reopen this pr either way, however, since it was originally going to be merged into the `general-cleanup` branch.

**Comment by tchoutri** [comment]
Yup' :) bonus point if you use their API! :)

**Comment by skwerlman** [comment]
i was able to get the nyaasi scraper 100% working, but nyaapantsu is down atm, so i don't know whether its scraper works (but its so similar to nyaasi that it should)

**Comment by tchoutri** [comment]
Great, thanks :)

**Comment by skwerlman** [comment]
closed in favor of #93

**Pull request closed without merging**